src/community/backup-restore/core/src/main/java/org/geoserver/backuprestore/tasklet/CatalogSecurityManagerTasklet.java

```diff
@@ -0,0 +1,169 @@
+                testGssm.reload();
+
+                // TODO: Perform validation tests here using "testGssm"
+
+                // TODO: Save warnings and validation issues on the JobContext
+            } catch (Exception e) {
+                logValidationExceptions((ValidationResult) null,
+                        new UnexpectedJobExecutionException(
+                                "Exception occurred while storing GeoServer security and services settings!",
+                                e));
+            } finally {
+                if (testGssm != null) {
+                    try {
+                        testGssm.destroy();
+                    } catch (Exception e) {
+                        LOGGER.log(Level.WARNING, "Test GeoServerSecurityManager Destry Error!", e);
+                    }
+                }
+            }
+        } else {
+            /**
+             * Create a new GeoServerSecurityManager instance using the INPUT DATA DIR.
+             * 
+             * Try to load the configuration from there and if everything is ok: 1. Replace the security folders 2. Destroy and reload the appContext
+             * GeoServerSecurityManager 3. Issue SecurityManagerListener extensions handlePostChanged(...)
+             */
+            final String inputFolderURL = jobExecution.getJobParameters()
+                    .getString(Backup.PARAM_INPUT_FILE_PATH);
+            Resource sourceRestoreFolder = Resources.fromURL(inputFolderURL);
+
+            // Test that the security folder has been correctly saved
+            GeoServerSecurityManager testGssm = null;
+            try {
+                testGssm = new GeoServerSecurityManager(
+                        new GeoServerDataDirectory(sourceRestoreFolder.dir()));
+                testGssm.setApplicationContext(Backup.getContext());
+                testGssm.reload();
+
+                // TODO: Perform validation tests here using "testGssm"
+
+                // TODO: Save warnings and validation issues on the JobContext
+            } catch (Exception e) {
+                logValidationExceptions((ValidationResult) null,
+                        new UnexpectedJobExecutionException(
+                                "Exception occurred while storing GeoServer security and services settings!",
+                                e));
+            } finally {
+                if (testGssm != null) {
+                    try {
+                        testGssm.destroy();
+                    } catch (Exception e) {
+                        LOGGER.log(Level.WARNING, "Test GeoServerSecurityManager Destry Error!", e);
+                    }
+                }
+            }
+
+            // Do this *ONLY* when DRY-RUN-MODE == OFF
+            if (!isDryRun()) {
+                // Copy the Security files into the destination folder
+                // TODO: Purge datadir option
+                Files.delete(security.dir());
```

Comment:
Currently there is only this implementation and the APIs do not envisage any other kind of security subsystem (unless I am missinig something, but I'm pretty sure that currently is like this).